### PR TITLE
Faster faster shape

### PR DESCRIPTION
**Pull request by kinnou02**
Shape were serialized then read multiple time in ed, they are now stored only one time in ed.

### CODE REVIEW

File: source/ed/data.h [hunk 1]
```diff
@@ -255,6 +256,10 @@ class Data: boost::noncopyable {
         for(ed::types::StopTime* stop : stops){
             delete stop;
         }
+
+        for(ed::types::Shape* shape : shapes_from_prev){
+            delete shape;
+        }
```
Comment (TeXitoi) [inline source/ed/data.h]: Please no, `new` and `delete` are now forbidden.

File: source/type/pt_data.cpp [hunk 2]
```diff
@@ -247,5 +247,9 @@ PT_Data::~PT_Data() {
     for (auto cal: associated_calendars) {
         delete cal;
     }
+
+    for (auto s: shapes) {
+        delete s;
+    }
```
Comment (TeXitoi) [inline source/type/pt_data.cpp]: `new` and `delete` are forbidden.

File: source/ed/data.cpp [hunk 3]
```diff
@@ -519,7 +520,9 @@ void Data::build_shape_from_prev() {
                     std::string key = (boost::format("%s|%s|%s") % vj->shape_id % prev_stop_point->uri % stop_time->stop_point->uri).str();
                     // we keep in cache the resulting geometry, so we don't have to re compute it later
                     if(shape_cache.find(key) == shape_cache.end()){
-                        shape_cache[key] = create_shape(prev_stop_point->coord, stop_time->stop_point->coord, shape.front());
+                        auto s = new types::Shape(create_shape(prev_stop_point->coord, stop_time->stop_point->coord, shape.front()));
```
Comment (pbougue) [inline source/ed/data.cpp]: length

File: source/ed/data.h [hunk 4]
```diff
@@ -255,6 +256,10 @@ class Data: boost::noncopyable {
         for(ed::types::StopTime* stop : stops){
             delete stop;
         }
+
+        for(ed::types::Shape* shape : shapes_from_prev){
```
Comment (pbougue) [inline source/ed/data.h]: spacing

File: source/ed/ed_persistor.cpp [hunk 5]
```diff
@@ -887,17 +891,37 @@ void EdPersistor::insert_validity_patterns(const std::vector<types::ValidityPatt
     this->lotus.finish_bulk_insert();
 }
 
+void EdPersistor::insert_shapes(const std::vector<types::Shape*>& shapes){
```
Comment (pbougue) [inline source/ed/ed_persistor.cpp]: spacing

File: source/ed/ed_persistor.cpp [hunk 6]
```diff
@@ -887,17 +891,37 @@ void EdPersistor::insert_validity_patterns(const std::vector<types::ValidityPatt
...
+    std::vector<std::string> columns = {"id", "geom"};
+
+    this->lotus.prepare_bulk_insert("navitia.shape", columns);
+    size_t inserted_count = 0;
+    std::vector<std::string> values;
+    for(types::Shape* shape: shapes){
```
Comment (pbougue) [inline source/ed/ed_persistor.cpp]: spacing

File: source/ed/ed_reader.cpp [hunk 7]
```diff
@@ -997,6 +998,17 @@ void EdReader::fill_meta_vehicle_journeys(nt::Data& data, pqxx::work& work) {
     }
 }
 
+void EdReader::fill_shapes(nt::Data& data, pqxx::work& work) {
+    std::string request = "SELECT id as id, ST_AsText(geom) as geom FROM navitia.shape";
+    const pqxx::result result = work.exec(request);
+    for(auto const_it = result.begin(); const_it != result.end(); ++const_it) {
```
Comment (pbougue) [inline source/ed/ed_reader.cpp:1004]: spacing

File: source/sql/alembic/versions/27fae61b6786_add_shape_table.py [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+from alembic import op
+import sqlalchemy as sa
+import geoalchemy2 as ga
+
+
+def upgrade():
+    op.create_table('shape',
+    sa.Column('id', sa.BIGINT(), nullable=False),
+    sa.Column('geom', ga.Geography(geometry_type='LINESTRING', srid=4326, spatial_index=False), nullable=True),
+    sa.PrimaryKeyConstraint('id'),
```
Comment (pbougue) [inline source/sql/alembic/versions/27fae61b6786_add_shape_table.py]: indent could be clearer

File: source/ed/ed_reader.cpp [hunk 9]
```diff
@@ -997,6 +998,17 @@ void EdReader::fill_meta_vehicle_journeys(nt::Data& data, pqxx::work& work) {
...
+            nt::LineString* shape = new nt::LineString();
```
Comment (pbougue) [inline source/ed/ed_reader.cpp]: why not pushing back directly an object?

File: source/ed/data.cpp (hunk 3 above)
Comment (kinnou02) [inline source/ed/data.cpp]: it look fine...

File: source/ed/ed_reader.cpp [hunk 10]
```diff
@@ -997,6 +998,16 @@ void EdReader::fill_meta_vehicle_journeys(nt::Data& data, pqxx::work& work) {
     }
 }
 
+void EdReader::fill_shapes(nt::Data& data, pqxx::work& work) {
+    std::string request = "SELECT id as id, ST_AsText(geom) as geom FROM navitia.shape";
+    const pqxx::result result = work.exec(request);
+    for(auto const_it = result.begin(); const_it != result.end(); ++const_it) {
+            auto shape = boost::make_shared<nt::LineString>();
```
Comment (pbougue) [inline source/ed/ed_reader.cpp]: indent

**Pull request merged**